### ISSUE TITLE
Close more files explicitly

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8589]
|[<img alt="scop's avatar" src="https://avatars.githubusercontent.com/u/109152?s=50" width="50" height="50">](https://github.com/scop)| @scop reported|
|-|-|
|Trac ID|trac#8589|
|Type|enhancement|
|Created|2016-07-12 10:21:10Z|
|Branch|https://github.com/twisted/twisted/tree/8589-scop-closing|

See extended description in [#8573](https://github.com/twisted/twisted/issues/8573)

<details><summary>Searchable metadata</summary>

```
trac-id__8589 8589
type__enhancement enhancement
reporter__scop scop
priority__normal normal
milestone__None None
branch__8589_scop_closing 8589-scop-closing
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1468318870781182 1468318870781182
changetime__1468603165314401 1468603165314401
version__None None
owner__scop scop

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="scop's avatar" src="https://avatars.githubusercontent.com/u/109152?s=50" width="50" height="50">](https://github.com/scop)<a name="note_1"></a>|@scop commented|
|-|-|

https://github.com/twisted/twisted/pull/316

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban set owner to @scop|
|-|-|

Here is the review.

From what I can see in the code, the known host part it only support the old format for the known hosts.

I would argue that such code should be either removed or fixed ... as right now it is in a zombie state 

The current coverage is more than good enough in the case in which in production users are only loading known host with old format and without any errors in it :)

I think that we can merge as it is.

Rather than trying to keep this zombie code I think that in the future we should put the effort in removing it.

Will look at merging the branch. I can handle the out of date stuff

Thanks!

**Comment by twisted-trac** [comment]
|<img alt="Ville Skyttä's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|Ville Skyttä commented|
|-|-|

In changeset 0f31378777fd6c243b8094e973176fdeaf9e30ff
```
#!CommitTicketReference repository="" revision="0f31378777fd6c243b8094e973176fdeaf9e30ff"
Add topfile for #8589
```

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban set status to `closed`|
|-|-|

In changeset 2dd6d509987b0ea3936e86e77225836b1eaaeaf4
```
#!CommitTicketReference repository="" revision="2dd6d509987b0ea3936e86e77225836b1eaaeaf4"
Merge 8589-scop-closing: Refactor more code to use open context for handling files.

Author: scop
Reviewer: adiroiban
Fixes: #8589
```
